lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -82,20 +82,22 @@ sub sqlt_deploy_hook ($self, $sqlt_table) {
 
 sub results ($self, %options) {
     my $skip_text_data = $options{skip_text_data};
+    my $errors = $options{errors};
 
     return {} unless my $dir = $self->job->result_dir;
     my $name = $self->name;
 
     my $file = path($dir, "details-$name.json");
     my $json_data = do {
         try { $file->slurp }
-        catch ($e) { }
+        catch ($e) { push @$errors, qq{Unable to read "$file": $e} if $errors }
+      }
+      or return {};
+    my $json = do {
+        try { decode_json($json_data) }
+        catch ($e) { push @$errors, qq{Malformed JSON file "$file": $e} if $errors }
```

Comment:
Hm, in the error case `$json` would end up with the result of the `push` command, usually `1`.
It seems that it doesn't make the tests fail, because if `$json == 1` then the
```
    if (ref $details ne 'ARRAY') {                                              
        $results->{details} = [];                                               
        return $results;                                                        
    }                                                                           
```
will return, but it would be nice if that would be handled earlier.